docs/source/development/styleguide.rst

```diff
@@ -89,3 +92,37 @@ Your contribution should fulfill the following criteria:
 - Don't use global variables unless absolutely necessary
     Exceptions are global variables from a config file that replace magic numbers.
     Never use mutable global variables!
+
+Styleguide for the documentation
+--------------------------------
+
+- General.
+    The documentation is rendered with `Sphinx <https://www.sphinx-doc.org/en/master/>`_
+    and  written in **reStructuredText.** How-to guides are usually Jupyter notebooks.
+
+- Purpose of documents.
+    Our documentation is inspired by the `system <https://documentation.divio.com/>`_
+    developed by Daniele Procida.
+
+      - How-to guides are problem-oriented and show how to achieved specific tasks.
+      - Explanations contain information on theoretical
+        concepts underlying estimagic, such as numerical differentiation and
+        moment-based estimation.
+      - The API Reference section contains auto-generated API reference
+        documentation and provides additional details about the implementation.
+
+- Headings.
+    Only the first letter of a title is capitalized. Moreover, use the following
+    order of headings:
+
+      - === (main heading)
+      - -\\-\\-
+      - ~~~
+      - ^^^
+      - """
+
+- Format.
```

Comment:
I think Tobi found a way to automate this with a pre-commit hook: https://github.com/covid-19-impact-lab/sid/blob/80379816d5502eec5070963076542e142e8afd61/.pre-commit-config.yaml#L50

Can you add that to estimagic as well (in pre-commit-config.yaml and environment.yaml)?